### PR TITLE
refactor(other): moved token expantion into style dictionary

### PR DESCRIPTION
**Pull request by daragh-king-genesys**
moved token expantion into style dictionary

✅ Closes: COMUI-3064

### CODE REVIEW

File: packages/genesys-spark-tokens/snapshot-new/json/gse-legacy-light-ui.json [hunk 1]
```diff
@@ -1345,7 +1345,7 @@
     "box_shadow": {
       "blur": "4px",
       "spread": "1px",
-      "color": "rgba(35,57,92,0.1)",
+      "color": "#23395c",
```
Comment (321gillian) [inline packages/genesys-spark-tokens/snapshot-new/json/gse-legacy-light-ui.json:1348]: These colours look different to me, is there a reason those are changing as part of this?

File: packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-ui.json [hunk 2]
```diff
@@ -198,7 +198,7 @@
     "padding": "24px",
     "borderless": {
       "border": {
-        "color": "transparent",
+        "color": "#000000",
```
Comment (321gillian) [inline packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-ui.json:201]: same here

File: packages/genesys-spark-tokens/snapshot-new/scss/gse-flare-dark-semantic.scss [hunk 3]
```diff
@@ -466,9 +730,24 @@
...
+  --gse-semantic-contanier-md-boxShadow-color: rgba(35, 57, 92, 0.12); /* The shadow we used for medium-sized components like elevated cards or menus. */
+  --gse-semantic-contanier-md-boxShadow-type: dropShadow; /* The shadow we used for medium-sized components like elevated cards or menus. */
+  --gse-semantic-contanier-md-boxShadow-offsetX: 0; /* The shadow we used for medium-sized components like elevated cards or menus. */
+  --gse-semantic-contanier-md-boxShadow-offsetY: 0; /* The shadow we used for medium-sized components like elevated cards or menus. */
+  --gse-semantic-contanier-sm-boxShadow-blur: 4px; /* The shadow we used for small-sized components like tooltips. */
+  --gse-semantic-contanier-sm-boxShadow-spread: 1px; /* The shadow we used for small-sized components like tooltips. */
+  --gse-semantic-contanier-sm-boxShadow-color: rgba(35, 57, 92, 0.1); /* The shadow we used for small-sized components like tooltips. */
+  --gse-semantic-contanier-sm-boxShadow-type: dropShadow; /* The shadow we used for small-sized components like tooltips. */
```
Comment (321gillian) [inline packages/genesys-spark-tokens/snapshot-new/scss/gse-flare-dark-semantic.scss:748]: Where is this dropShadow defined?

File: packages/genesys-spark-tokens/snapshot-new/json/gse-legacy-light-ui.json (hunk 1 above)
Comment (daragh-king-genesys) [reply packages/genesys-spark-tokens/snapshot-new/json/gse-legacy-light-ui.json]: Fixed

File: packages/genesys-spark-tokens/snapshot-new/scss/gse-flare-dark-semantic.scss (hunk 3 above)
Comment (daragh-king-genesys) [reply packages/genesys-spark-tokens/snapshot-new/scss/gse-flare-dark-semantic.scss:748]: Fixed

File: packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-ui.json (hunk 2 above)
Comment (daragh-king-genesys) [reply packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-ui.json]: Fixed

File: packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-core.json [hunk 4]
```diff
@@ -159,16 +159,16 @@
...
+      "yellow_green": "#f8f3c6",
+      "olive_green": "#dcdbbb",
+      "aqua_green": "#c7e5e8",
+      "pink": "#ffdff5",
+      "fuchsia": "#f4c9ec",
+      "lilac": "#e9e8f9",
+      "plum": "#cdc9d8",
+      "electric_purple": "#dcc9f2",
+      "genesys_blue": "#d9e4ff",
+      "navy": "#bbbfd4"
```
Comment (321gillian) [inline packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-core.json:171]: These are much lighter versions of the current colours. Just confirming that was intended?

File: packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-semantic.json [hunk 5]
```diff
@@ -24,7 +24,7 @@
           "30": "#252e78"
         },
         "tertiary": {
-          "10": "#000000",
+          "10": "rgba(0, 0, 0, 0)",
```
Comment (321gillian) [inline packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-semantic.json:27]: This is going from black to transparent. Is that correct?

File: packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-semantic.json [hunk 6]
```diff
@@ -41,7 +41,7 @@
         }
       },
       "overlay": {
-        "10": "#3e5374"
+        "10": "#3e5374cc"
```
Comment (321gillian) [inline packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-semantic.json:44]: This is too long to be a valid hex code I think

File: packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-ui.json [hunk 7]
```diff
@@ -1473,7 +1473,7 @@
       "default_color": "#3e5374",
       "hover_color": "#23395d",
       "active_color": "#1b2c48",
-      "disabled_color": "#3e5374"
+      "disabled_color": "#3e537480"
```
Comment (321gillian) [inline packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-ui.json:1476]: Hex code looks too long here too

File: packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-core.json (hunk 4 above)
Comment (daragh-king-genesys) [reply packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-core.json:171]: 99% sure this change is correct. Will be able to confirm 100% once we start testing with UX

File: packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-semantic.json (hunk 5 above)
Comment (daragh-king-genesys) [reply packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-semantic.json:27]: Yes it was wrong before. ie the json files were missing their alpha channel info before this change.

File: packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-semantic.json (hunk 6 above)
Comment (daragh-king-genesys) [reply packages/genesys-spark-tokens/snapshot-new/json/gse-flare-dark-semantic.json:44]: The extra characters are for the alpha channel.
https://caniuse.com/css-rrggbbaa
https://developer.mozilla.org/en-US/docs/Web/CSS/hex-color

**Review by 321gillian — APPROVED** [review]

**Review by 321gillian — CHANGES_REQUESTED** [review]
Spotted a visual difference. Just removing approval until I discuss it with you

**Review by 321gillian — APPROVED** [review]

**Pull request merged**